3/ArrayDuplicates.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AgileFreaks_test1
+{
+    public class ArrayDuplicates
+    {
+        public bool hasDuplicates(int[] array)
+        {
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (array[array[i] % array.Length] > array.Length)
```

Comment:
You are doing this calculation twice `array[i] % array.Length` you can extract in a variable